### ISSUE TITLE
How to save created image in unity-webview?

### ISSUE THREAD

**Issue by rokrodic** [gree/unity-webview#874]
Hi. I have the following javascript code. I do some drawing onto canvas... and at the end I create a download link and force download image. I cannot find the file on android device...

`<script>`
`var cA=document.getElementById('canAvg');`
`var dtxA=cA.getContext("2d");`
`dtxA.fillText("30",0,50);`
`var a = document.createElement('a');`
`a.href = cA.toDataURL("image/png");`
`a.download = 'my_file_name.png';`
`a.click();`
`</script>`

How do I get the image from webview? If the above is not the right way - how do I get it? Simple way.

(BTW. I know I can use something like: 
`var pas1={img:cA.toDataURL("image/png"),name:'filename.png'}`
`var query1=[];for (var key in pas1) {query1.push(encodeURIComponent(key)+'='+encodeURIComponent(pas1[key]));}`
`myPost('https://link/some_upload.php',query1.join('&'));`
and have myPost use `XMLHttpRequest();...` to post the image onto a server... and then request the image back to the app, but this is way to dependent and complicated...)

Thanks!

**Comment by KojiNakamaru** [comment]
Currently the plugin doesn't support file downloading and its implementation would be a bit tricky/unreliable. Instead, I added a new (experimental) javascript function `Unity.saveDataURL(fileName, dataURL)` in #875 and updated binaries. You can use it as below:

```javascript
var cA = document.getElementById('canAvg');
var dtxA = cA.getContext("2d");
dtxA.fillText("30",0,50);
var a = document.createElement('a');
a.href = cA.toDataURL("image/png");
a.download = 'my_file_name.png';
//a.click();
Unity.saveDataURL(a.download, a.href);
```

**Comment by rokrodic** [comment]
Hi. Great, it works! I have tested it.

Only drawback is it displays a window asking where to save the file. It would be nice to specify this in advance as a parameter. Is it possible to do it incognito?
Other drawback is it downloads/saves only one image. I have:
`var a3=document.createElement('a');a3.href=cA.toDataURL(""image/png"");a3.download='img1.png';Unity.saveDataURL(a3.download, a3.href);
var a4=document.createElement('a');a4.href=cM.toDataURL(""image/png"");a4.download='img2.png';Unity.saveDataURL(a4.download, a4.href);
`
I only get img1.png saved. I try to save 3 images. Maybe I have to do it with a delay? I have android version 13.

Anyways, the saveDataURL works and thank you!

**Comment by KojiNakamaru** [comment]
I modified the code in #876 and updated binaries. Could you please try the latest.

**Comment by rokrodic** [comment]
Hi. Update works. Downloads are made into download folder. Thanks!

**Comment by rokrodic** [comment]
Hi Koji!
As I have written above, the "saveDataURL" you have added to Android works flowlessly ([link](https://github.com/gree/unity-webview/blob/5deeb5ef40c725ed914f6c8044afef636c393e74/plugins/Android/webview/src/main/java/net/gree/unitywebview/CWebViewPlugin.java#L118)).

Now I am transferring the code to IOS and would kindly request to add this feature to IOS too. Can you?

**Comment by rokrodic** [comment]
Hi Koji!
Any progress?

**Comment by KojiNakamaru** [comment]
I implemented saveDataURL for iOS in #904 and updated binaries. Please check #904 for a limitation about Downloads folder.

**Comment by rokrodic** [comment]
For those who search for the IOS solution follow the #904 link.
